### ISSUE TITLE
ein:console-open fails to work: if: python.el is not loaded!

### ISSUE THREAD

**Issue by yarikoptic** [tkf/emacs-ipython-notebook#91]
I do have python.el loaded (remove python-mode for now entirely), started with zeroin.el.  C-c C-o leads to this error while in _Messages_ I have

```
if: python.el is not loaded!
Loading python...
ad-handle-definition: `python-eldoc-function' got redefined
Loading python...done
if: python.el is not loaded!
Mark set
```

as you see above -- I did load python.el again, but C-c C-o failed anyways with the same msg

**Comment by tkf** [comment]
Right, you need the new python.el (https://github.com/fgallina/python.el).  The message is not good.  I need to change it.

You can either upgrade to Emacs 24.2 or later.  If you want to keep using Emacs 23.x, you can download 23-compatible python.el from here: https://github.com/fgallina/python.el/tree/emacs23 (it is in emacs23 branch).

**Comment by yarikoptic** [comment]
ah -- thanks!  updating my emacs to ... 45.0

my ultimate wish/goal, actually, is to make EIN work with what we ship in Debian.  Currently there is a fresh releases of python-mode getting cooked up quite often and fresh one (6.1.0) will appear in Debian sid some time very soon.  Have you looked at those as well? (first Ihave tried to start EIN in my .emacs-settings with that python-mode but it felt broken in many ways)

**Comment by yarikoptic** [comment]
damn ... emacs 45.0 is just a version of the meta package (not sure why that number) depending on emacs23 -- co-installing 24 now. thanks

**Comment by yarikoptic** [comment]
"the new python.el (https://github.com/fgallina/python.el)"  -- do you know if this one would ever become a part of GNU Emacs?  this proliferation of python*.el's is at least disturbing... and this one is not what is the actually "new ipython.el" in emacs24 -- the name should have been chosen different from the stock one... grrr

Since this one is not a standard one -- why not to add it to lib/ ?

**Comment by tkf** [comment]
I was not clear about it.  Yes, it is part of GNU Emacs now, as of Emacs 24.2.  The README is bit old and actually the central repository is now Emacs's bzr repository.  But the github issue tracker is active.

**Comment by yarikoptic** [comment]
it must be that emacs 24 updated  to this python.el later than 24.2:

```
revno: 108614 [merge]
committer: Fabián Ezequiel Gallina <fabian@anue.biz>
branch nick: trunk
timestamp: Thu 2012-06-14 23:43:05 -0300
message:
  Merge new python.el
```

I got original bzr of emacs, 24.2 release does contain the same python.el as in Debian package of 24.2 (I was already about to complain to Debian maintainers ;) ), and it is no good for EIN's "console".

But even by loading shipped in lib/python.el one -- it fails to open a console for me due to

```
[ZMQTerminalIPythonApp] Could not find existing kernel connection file kernel-020a843d-62ac-479d-b920-13ab32f0a8fd.json

Process Python[*ein: 8888/fs_pandas_analysis*] exited abnormally with code 1
```

my emacs with EIN was started in EIN source code base while ipython notebook server was opened in other directory (if that is of relevance).

P.S. use of bzr for emacs codebase is my current definition for "masochism"

**Comment by tkf** [comment]
I was just asked the same question here: https://github.com/tkf/emacs-ipython-notebook/issues/86#issuecomment-11099859

Yea, building from bzr is scary.
